tools/coredump/modulestore/store.go

```diff
@@ -62,12 +63,26 @@ type Store struct {
 
 // New creates a new module storage. The modules present in the local cache are inspected and a
 // full index of the modules in the remote S3 bucket is retrieved and cached as well.
-func New(s3client *s3.Client, s3Bucket, localCachePath string) *Store {
+func New(s3client *s3.Client, s3Bucket, localCachePath string) (*Store, error) {
+	fi, err := os.Stat(localCachePath)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			if err = os.Mkdir(localCachePath, 0750); err != nil {
+				return nil, err
+			}
```

Comment:
nit:
Normally, I'd do this

mkdir() unconditionally, if OK return ok
if err != EEXISTS return error
now check with stat() that it is a directory ...

Reply:
Or even briefer: use `os.MkdirAll` which doesn't return an error for existing directory.